src/diffusers/loaders.py

```diff
@@ -962,72 +1047,68 @@ def lora_scale(self) -> float:
+                attn_module.q_proj = PatchedLoraProjection(attn_module.q_proj, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.q_proj.lora_linear_layer.parameters())
 
-                    # create a new scope that locks in the old_forward, lora_layer value for each new_forward function
-                    # for more detail, see https://github.com/huggingface/diffusers/pull/3490#issuecomment-1555059060
-                    def make_new_forward(old_forward, lora_layer):
-                        def new_forward(x):
-                            result = old_forward(x) + self.lora_scale * lora_layer(x)
-                            return result
+                attn_module.k_proj = PatchedLoraProjection(attn_module.k_proj, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.k_proj.lora_linear_layer.parameters())
 
-                        return new_forward
+                attn_module.v_proj = PatchedLoraProjection(attn_module.v_proj, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.v_proj.lora_linear_layer.parameters())
 
-                    # Monkey-patch.
-                    module.forward = make_new_forward(old_forward, lora_layer)
+                attn_module.out_proj = PatchedLoraProjection(attn_module.out_proj, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.out_proj.lora_linear_layer.parameters())
+            elif isinstance(text_encoder, T5EncoderModel):
+                attn_module.q = PatchedLoraProjection(attn_module.q, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.q.lora_linear_layer.parameters())
 
-    @property
-    def _lora_attn_processor_attr_to_text_encoder_attr(self):
-        return {
-            "to_q_lora": "q_proj",
-            "to_k_lora": "k_proj",
-            "to_v_lora": "v_proj",
-            "to_out_lora": "out_proj",
-        }
+                attn_module.k = PatchedLoraProjection(attn_module.k, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.k.lora_linear_layer.parameters())
+
+                attn_module.v = PatchedLoraProjection(attn_module.v, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.v.lora_linear_layer.parameters())
+
+                attn_module.o = PatchedLoraProjection(attn_module.o, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.o.lora_linear_layer.parameters())
+            else:
+                raise ValueError(f"{text_encoder.__class__.__name__} does not support LoRA training")
 
-    def _load_text_encoder_attn_procs(
-        self, pretrained_model_name_or_path_or_dict: Union[str, Dict[str, torch.Tensor]], **kwargs
+        return lora_parameters
+
+    @classmethod
+    def _load_text_encoder_lora_state_dict(
+        cls, pretrained_model_name_or_path_or_dict: Union[str, Dict[str, torch.Tensor]], **kwargs
     ):
```

Comment:
This now returns a state dict which can be just be loaded via `torch.load_state_dict`

Reply:
never mind can just delete this function